smartcheck/check.py

```diff
@@ -33,15 +35,43 @@
 
 TEST_RESULT_RE = re.compile(r"#\s*(\d+)\s+(.*?)\s{2,}(.*?)\s{2,}\s+([\d%]+)\s+(\d+)\s+(\d+|-)", re.UNICODE)
 
+INFORMATION_RE_NVME = [
+    ('model_number', re.compile(r'Model Number:\s*(.*)', re.UNICODE)),
+    ('serial', re.compile(r'Serial Number: (.*)', re.UNICODE)),
+    ('firmware_version', re.compile(r'Firmware Version: (.*)', re.UNICODE)),
+]
+
+# SMART data attributes regex from nvme
+
+DATA_ATTRIBUTES_RE_NVME = [
+    ('critical_warning', re.compile(r'Critical Warning:\s*(0x[0-9a-f]{2})', re.UNICODE)),
+    ('available_spare', re.compile(r'Available Spare:\s*(\d+)%', re.UNICODE)),
+    ('available_spare_treshold', re.compile(r'Available Spare Threshold:\s*(\d+)%', re.UNICODE)),
+    ('unsafe_shutdowns', re.compile(r'Unsafe Shutdowns:\s*([0-9\.,]+)', re.UNICODE)),
+    ('media_and_data_integrity_errors', re.compile(r'Media and Data Integrity Errors:\s*([0-9\.,]+)', re.UNICODE)),
+    ('error_information_log_entries', re.compile(r'Error Information Log Entries:\s*([0-9\.,]+)')),
+    ('warning_comp_temperature_time', re.compile(r'Warning  Comp\. Temperature Time:\s*([0-9\.,]+)')),
+    ('critical_comp_temperature_time', re.compile(r'Critical Comp\. Temperature Time:\s*([0-9\.,]+)'))
+]
+
 
 def toint(s, default=0):
     try:
         return int(s)
     except ValueError:
-        return default
+        try:
+            return int(s, 16)
```

Comment:
This may silently return the wrong value for strings that aren't integers in base 10 but happen to be valid in base 16. Please add the base as an optional kwarg and add the base at the call sites.

Reply:
Thank you for the review! I fixed this.